### PR TITLE
sd-id128: avoid an unnecessary function call in inline helper

### PR DESCRIPTION
**Pull request by keszybz**
When optimizing, the compiler will most likely replace the call to memcmp(),
but at -O0, the code that is emitted builds the call preamble and does the
call. Let's use the same pattern as with sd_id128_is_null() and
sd_id128_is_allf() and avoid the call.

Split out of #23773, since apparently it's making the compiler unhappy.

### CODE REVIEW

**Comment by yuwata** [comment]
```
In file included from ../src/systemd/sd-bus.h:27,
                 from ../src/journal/journalctl.c:21:
In function â€˜sd_id128_equalâ€™,
    inlined from â€˜mainâ€™ at ../src/journal/journalctl.c:2697:46:
../src/systemd/sd-id128.h:119:43: error: â€˜previous_boot_id.qwords[0]â€™ may be used uninitialized [-Werror=maybe-uninitialized]
  119 |         return a.qwords[0] == b.qwords[0] && a.qwords[1] == b.qwords[1];
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/journal/journalctl.c: In function â€˜mainâ€™:
../src/journal/journalctl.c:2167:20: note: â€˜previous_boot_id.qwords[0]â€™ was declared here
 2167 |         sd_id128_t previous_boot_id;
      |                    ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

File: src/home/homework-luks.c [hunk 1]
```diff
@@ -1229,7 +1229,7 @@ int home_setup_luks(
                 PasswordCache *cache,
                 UserRecord **ret_luks_home) {
 
-        sd_id128_t found_partition_uuid = SD_ID128_NULL, found_luks_uuid = SD_ID128_NULL, found_fs_uuid = SD_ID128_NULL;
+        sd_id128_t found_partition_uuid, found_luks_uuid, found_fs_uuid;
```
Comment (yuwata) [inline src/home/homework-luks.c]: `found_luks_uuid` may not be initialized, as `luks_open()` is called only when `!setup->crypt_device`.

**Review by yuwata — CHANGES_REQUESTED** [review]
The CI failure seems related.

File: src/home/homework-luks.c (hunk 1 above)
Comment (keszybz) [reply src/home/homework-luks.c]: Arggh, indeed. Should be better now.

**Pull request merged**